server/utils/typeHelpers.ts

```diff
@@ -8,17 +9,35 @@ import type {
 } from '@server/api/themoviedb/interfaces';
 
 export const isMovie = (
-  movie: TmdbMovieResult | TmdbTvResult | TmdbPersonResult
+  movie:
+    | TmdbMovieResult
+    | TmdbTvResult
+    | TmdbPersonResult
+    | TmdbCollectionResult
 ): movie is TmdbMovieResult => {
   return (movie as TmdbMovieResult).title !== undefined;
 };
 
 export const isPerson = (
-  person: TmdbMovieResult | TmdbTvResult | TmdbPersonResult
+  person:
+    | TmdbMovieResult
+    | TmdbTvResult
+    | TmdbPersonResult
+    | TmdbCollectionResult
 ): person is TmdbPersonResult => {
   return (person as TmdbPersonResult).known_for !== undefined;
 };
 
+export const isCollection = (
+  collection:
+    | TmdbMovieResult
+    | TmdbTvResult
+    | TmdbPersonResult
+    | TmdbCollectionResult
+): collection is TmdbCollectionResult => {
+  return (collection as TmdbCollectionResult).title !== undefined;
```

Comment:
This won't work correctly since that condition is also valid for a `TmdbMovieResult` object. 

Unfortunately for us, all the properties of a `TmdbCollectionResult` are also present in a `TmdbMovieResult` and are of the same type - the only difference being the `media_type`. Hence, this should be updated to check the `media_type` instead.
```suggestion
  return (collection as TmdbCollectionResult).media_type === 'collection';
```